cpp/src/sort/sort_column.cu

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sort/sort_impl.cuh>
+
+namespace cudf {
+namespace detail {
+namespace {
+
+/**
+ * @brief Type-dispatched functor for sorting a single column.
+ */
+struct column_sorted_order_fn {
+  /**
+   * @brief Sorts fixed-width columns using faster thrust sort.
+   *
+   * @param input Column to sort
+   * @param indices Output sorted indices
+   * @param ascending True if sort order is ascending
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   */
+  template <typename T, typename std::enable_if_t<cudf::is_fixed_width<T>()>* = nullptr>
+  void faster_sort(column_view const& input,
```

Comment:
I think it'd be clearer and better express intent to explicitly call this `radix_sort` instead of `faster_sort`.